lib/logidze/ignore_log_data.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module Logidze
+  # Add `has_logidze` method to AR::Base
+  module IgnoreLogData
+    extend ActiveSupport::Concern
+
+    require "logidze/ignored_columns" if Rails::VERSION::MAJOR < 5
+
+    # Fixes unexpected behavior (see more https://github.com/rails/rails/pull/34528):
+    # instead of using a type passed to `attribute` call, ignored column uses
+    # a type coming from the DB (in this case `.log_data` would return a plain hash
+    # instead of `Logidze::History`)
+    module AttributePatch
+      if Rails::VERSION::MAJOR < 5
+        def log_data
+          @log_data ||= Logidze::History.load(super)
+        end
+
+        def log_data!
+          @log_data ||= Logidze::History.load(super)
+        end
+      else
+        def log_data
+          @log_data ||= Logidze::History::Type.new.cast_value(super)
+        end
+      end
+    end
+
+    included do
+      include AttributePatch
+
+      self.ignored_columns += ["log_data"]
+    end
+
+    module ClassMethods # :nodoc:
+      def with_log_data
+        select(attribute_names + ["log_data"])
+      end
+    end
+
+    def log_data!
+      @log_data ||= self.class.where(id: id).pluck(:log_data).first
```

Comment:
Should we store the value in the `attributes["log_data"]`? As I understand, the following code reloads `log_data` (even if it present):

```ruby
user = User.with_log_data.find(id)
user.log_data!
```

TBH, the API looks like a hack.

Maybe, it's better to add a method to reload `log_data` explicitly instead? Smth like:

```ruby
def reload_log_data
  attributes["log_data"] = self.class.where(id: id).pluck(:log_data).first
end
```

This method could be a part of `Logidze::Model`.

Btw, we should not rely on `id` but on model's primary key instead:

```ruby
self.class.where(self.class.primary_key => send(self.class.primary_key)).pluck(:log_data).first
```

Reply:
Looks like `id` [takes care](https://github.com/rails/rails/blob/master/activerecord/lib/active_record/attribute_methods/primary_key.rb#L18) about resolving primary key

Reply:
It takes care of the value; does it take care of the key? (i.e. `where(id: ...)` still uses `id`)

Let's rename this method to `reload_log_data`, move to `Logidze::Model` and add a readme section. This method could also be useful to fetch the actual `log_data` after creating or updating a record.